src/control/mod.rs

```diff
@@ -70,6 +70,16 @@ pub fn from_u32<T: ResourceHandle>(raw: u32) -> Option<T> {
     RawResourceHandle::new(raw).map(T::from)
 }
 
+unsafe fn transmute_vec<T, U>(from: Vec<T>) -> Vec<U> {
```

Comment:
Isn't it more common to use the `safe_transmute` crate for these?

Reply:
I suppose that would warrant converting the whole crate to use such a crate (and/or [getting rid of the transmutes in the first place...](https://github.com/Smithay/drm-rs/issues/117#issuecomment-1159784840)). I assumed since this is very limited in scope and has an explicit example in the std docs I would just implement it right there.

Indeed, it is the same code as some other crates provide, excluding size and alignment checks, which might be a benefit of an existing crate (or it can be added to this function):
* https://docs.rs/transvec/latest/src/transvec/lib.rs.html#511
* https://docs.rs/safe-transmute/latest/src/safe_transmute/full.rs.html#298
* https://docs.rs/bytemuck/latest/src/bytemuck/allocation.rs.html#182

Personally I'm partial to bytemuck as it is most popular, but also the heaviest :woman_shrugging:

Reply:
I think that is out of the scope of this PR. Fixing the currently unsound behaviour should be done first, we can improve upon that later. Feel free to open another issue for discussing the different approaches.